src/Chapter3.hs

```diff
@@ -480,6 +511,51 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data City = City
+    { cityCastle  :: Maybe Castle
+    , cityWall    :: Maybe Wall
```

Comment:
While this is correct, however, this allows having unrepresentable states.
For example, nothing stops us from having `Just Wall` and `Nothing :: Maybe Castle`.

You can actually avoid that by using more flexible structures that precisely would describe all possible states in types and won't let us to create invalid ones.

For example, here, instead of having `Wall` and `Castle` separately, you can think on the may to create a data type, that would represent All: none cattle, castle and castle with walls 🙂